### PR TITLE
Hashsum download docs

### LINKED ISSUE #1651: How do you use the hashsum_download api?
I'm trying to use girder as a midas replacement/cmake externalData module storage. @mgrauer mentioned this could be done using the `hashsum_download` plugin, but there doesn't seem to be any documentation for it. 

Could you guys share an example curl command or something showing how to use it?

### PR DESCRIPTION
**Pull request by mgrauer**
Belatedly closes #1651 .

/cc @thewtex

### CODE REVIEW

**Comment by thewtex** [comment]
Awesome! Thanks @mgrauer ! Will review next week.

File: docs/plugins.rst [hunk 1]
```diff
@@ -491,3 +491,77 @@ users should now be able to run the task represented by the docker image.
...
+`ITKExamples <https://itk.org/ITKExamples/Documentation/Contribute/UploadBinaryData.html>`_. This
+example project uses the Girder instance https://data.kitware.com.
+
+.. note:: The use of the hashsum_download plugin with CMake ExternalData is only supported with a
+   filesystem assetstore and SHA512 as the hash algorithm.
+
+As every local Git repository contains a copy of the entire project history, it is important to
+avoid adding large binary files directly to the repository. Large binary files added and removed
+throughout a project’s history will cause the repository to become bloated and take up too much
+disk space, requiring excessive time and bandwidth to download, etc.
```
Comment (zachmullen) [inline docs/plugins.rst]: etc.?

File: docs/plugins.rst [hunk 2]
```diff
@@ -491,3 +491,77 @@ users should now be able to run the task represented by the docker image.
...
+
+A solution to this problem, when using the CMake build system, is to store binary files in a
+separate location outside the Git repository, then download the files at build time with CMake.
+
+A "content link" file contains an identifying SHA512 hash. The content link is calculated based on
```
Comment (zachmullen) [inline docs/plugins.rst]: In the GUI we call this a "key file", which could be a bad name. Let's either change that in the GUI or here in the docs so it's consistent.

File: docs/plugins.rst [hunk 3]
```diff
@@ -491,3 +491,77 @@ users should now be able to run the task represented by the docker image.
...
+Usage by a software project maintainer
+**************************************
+
+Again, for more background, using the example Girder instance https://data.kitware.com, see the
+`ITKExamples <https://itk.org/ITKExamples/Documentation/Contribute/UploadBinaryData.html>`_. Also
+see the CMake External Data documentation for CMake project configuration
+`help <https://cmake.org/cmake/help/latest/module/ExternalData.html>`_.
+
+In your project, you must set ExternalData_URL_TEMPLATES to a girder url, e.g.
+"https://data.kitware.com:443/api/v1/file/hashsum/%(algo)/%(hash)/download".
```
Comment (zachmullen) [inline docs/plugins.rst]: Let's leave off ports from URLs when they are the default value.

File: docs/plugins.rst [hunk 4]
```diff
@@ -491,3 +491,77 @@ users should now be able to run the task represented by the docker image.
...
+release and on a regular basis, perhaps nightly, the data should be archived in a new Girder folder
+to ensure its persistence.  A `script <https://github.com/InsightSoftwareConsortium/ITK/blob/ef14cce1c26d5dce7eb2e10d36c7dc81aaa9c9e8/Utilities/Maintenance/ArchiveTestingDataOnGirder.py>`_ that provides this functionality is available, as is an
+`example folder <https://data.kitware.com/#collection/57b5c9e58d777f126827f5a1/folder/57b672b48d777f10f269651a>`_
+produced by the script for a release.
+
+Usage by a software project contributor
+***************************************
+
+Upload a file to a Girder instance, which will create a Girder Item to house the file. Navigate to
+the Item, then click on the i (information) icon next to the file, which will show the id, and
```
Comment (zachmullen) [inline docs/plugins.rst]: Let's make this i bold or something

File: docs/plugins.rst [hunk 5]
```diff
@@ -491,3 +491,77 @@ users should now be able to run the task represented by the docker image.
...
+since the hashsum_download plugin is enabled, the sha512 field will also be displayed. Click on the
+key icon to download a hashfile, which will be the full sha512 of the file, with the same name as
```
Comment (zachmullen) [inline docs/plugins.rst]: Again for consistency, let's use either key file or content link file here instead of hashfile.

File: docs/plugins.rst [hunk 6]
```diff
@@ -491,3 +491,77 @@ users should now be able to run the task represented by the docker image.
...
+the file, and an extension of .sha512, and you can use this for your CMake content link. E.g.,
+upload my_datafile.txt and download the my_data.txt.sha512 file, then check the my_data.txt.sha512
+file into your source repository.
+
+You can use the Girder API to get the hash of the file given the file id, with the endpoint
+api/v1/file/<file id>/hashsum_file/sha512, where the file id comes from the specific file in Girder.
```
Comment (zachmullen) [inline docs/plugins.rst]: Let's format these with double backticks (code).

File: docs/plugins.rst [hunk 7]
```diff
@@ -491,3 +491,77 @@ users should now be able to run the task represented by the docker image.
...
+
+You can also use the API to download the file based on the hash returned by the previous endpoint,
+with an endpoint /api/v1/file/hashsum/sha512/<file sha512 hash>/download, where the sha512 hash
```
Comment (zachmullen) [inline docs/plugins.rst]: Same for this API endpoint.

File: docs/plugins.rst (hunk 2 above)
Comment (mgrauer) [inline docs/plugins.rst]: I will describe it as a key file in these docs, and mention that it is the same as what CMake considers a content link file.

**Comment by mgrauer** [comment]
Feedback addressed.

**Comment by thewtex** [comment]
Excellent improvements! This should help a lot of folks use Girder and CMake. Thanks @mgrauer and @zachmullen ! :1st_place_medal:

**Pull request merged**